include/lf.h

```diff
@@ -62,17 +62,12 @@ typedef struct {
 } LF_PINBOX;
 
 typedef struct {
-  void * volatile pin[LF_PINBOX_PINS];
+  void * volatile pin[LF_PINBOX_PINS] MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE);
   LF_PINBOX *pinbox;
   void  **stack_ends_here;
   void  *purgatory;
   uint32 purgatory_count;
   uint32 volatile link;
-/* we want sizeof(LF_PINS) to be 128 to avoid false sharing */
-  char pad[128-sizeof(uint32)*2
-              -sizeof(LF_PINBOX *)
-              -sizeof(void*)
-              -sizeof(void *)*(LF_PINBOX_PINS+1)];
 } LF_PINS;
```

Comment:
Here MY_ALIGNED() specifies alignment against the beginning of the structure, not the beginning of the memory. The only effect MY_ALIGNED() has here is sizeof(LF_PINS) now always equals to CPU_LEVEL1_DCACHE_LINESIZE

Since we store LF_PINS on unaligned malloc'ed memory region, this change makes LF_PINS vulnerable to false sharing.

Ideas (some mutual exclusive, some depend on each other):
- use aligned alloc (then we can avoid padding overhead)
- restore padding, but replace 128 with CPU_LEVEL1_DCACHE_LINESIZE * 2
- may be move MY_ALIGNED() down to "} MY_ALIGNED() LF_PINS;"
- replace MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE) with MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE * 2)